Production Release 2025/06/22

cfdxkk opened the pull request:
✨ 大量新功能与问题修复。
合并 `develop` 分支中的累积提交至 `production` 分支，发布最新版本。
@Aira-Sakuranomiya @otomad @YasaitoriOvO 

🚧 强依赖于某些后端 API，请前往 [预生产环境](https://stg-cerasus.kirafile.com/) 测试，或使用本地后端的 `develop` 分支测试。
🚫 在充分测试或代码状态达到最优前禁止合并。


#### 测试焦点
* KIRAKIRA RBAC
   * 该功能应该为无感更新，用户不应感受到任何区别。
      （涉及到 API 无法访问或奇怪报错时，请报告该问题）
* 关注
   * 关注或取消关注一个用户。
      * 各个页面中的关注 / 取关 按钮显示或交互是否正确。
* 黑名单
   * 用户屏蔽
      * 搜索并屏蔽一个用户。
      * 解除屏蔽用户。
      * 被屏蔽的用户投稿的视频不应在任何视频列表中出现。
      * 无法查看被屏蔽的用户的视频。
      * 无法进入被屏蔽的用户的主页。
      * 被屏蔽的用户的评论不应显示。
      * 被屏蔽的用户的弹幕不应显示。
      * 被其他用户屏蔽后，屏蔽你的用户投稿的视频不应在任何视频列表中出现。
      * 被其他用户屏蔽后，无法查看屏蔽你的用户的视频。
      * 被其他用户屏蔽后，无法进入屏蔽你的用户的主页。
      * 被其他用户屏蔽后，屏蔽你的用户的评论不应显示。
      * 被其他用户屏蔽后，屏蔽你的用户的弹幕不应显示。
   * 用户隐藏
      * 搜索并隐藏一个用户。
      * 显示用户。
      * 被隐藏的用户投稿的视频不应在任何视频列表中出现。
      * 被隐藏的用户的评论不应显示。
      * 被隐藏的用户的弹幕不应显示。
   * TAG 屏蔽
      * 添加屏蔽 TAG。
      * 删除屏蔽 TAG。
      * TAG 被屏蔽后，包含被屏蔽的 TAG 的视频不应在任何视频列表中出现。（？）
   * 关键词屏蔽
      * 添加屏蔽关键词。
      * 删除屏蔽关键词。
      * 不应添加正则表达式。
      * 视频标题中包含被屏蔽的关键词的视频不应在任何视频列表中出现。（？）
      * 包含被屏蔽的关键词的评论不应在任何视频评论中出现。
      * 包含被屏蔽的关键词的弹幕不应在任何视频弹幕中出现。
   * 正则表达式屏蔽
      * 添加屏蔽正则表达式。
      * 删除屏蔽正则表达式。
      * 不应添加危险正则表达式。
      * 视频标题能被屏蔽的关键词匹配的视频不应在任何视频列表中出现。（？）
      * 能被屏蔽的关键词匹配的评论不应在任何视频评论中出现。
      * 能被屏蔽的关键词匹配的弹幕不应在任何视频弹幕中出现。
* 播放器相关 UI 更新
   * 播放器周围的 UI 位置与样式调整。
* 本地化更新
   * 原则上全部文本（除 UGC 外）都应该支持本地化。
* 滚动条
   * 使用“受控的”滚动条组件，而不是浏览器实现的原生滚动条。
* 日期选择器
   * 实现日期选择器的基础功能。 @otomad 
* 其他 QoL 与安全更新

#### 已知问题
* 黑名单 - 使用 TAG，关键词，正则表达式方式屏蔽的视频，仍然会出现在用户主页的“视频”列表中（？需要修复吗）
* TAG 编辑器 - 在输入一个文字后，光标出现在 TAG 的最前方
* TAG 编辑器 - 在输入时使用输入法会导致出现重复文字
* 视频页 - SSR - 在视频页刷新会导致报错
* 在 iOS26 中，带有“液态玻璃”效果的 Safari 浏览器无法在底部显示内容。
* 无法更新用户生日，原因是与后端接口类型不一致。 @otomad 

#### 未包含功能
以下功能未能包含在本次更新中
* 隐私设置
   * 虽然隐私设置已经可以持久化保存，但该功能无实际效果，并且在未来会有较大的功能改变。
* 动态 / 动态分组

该文档仍然编辑中，可能存在错误或遗漏。
最后编辑时间：[2025/08/08]
</br>

---

</br>
❤按照惯例，为春季（夏季）大版本更新附加樱花图片。
<details> <summary>查看图片</summary>

![3003778ef9e3b3622f34ec27d20af8e7_720](https://github.com/user-attachments/assets/34dc0d73-5102-43aa-bd96-fa71ccfadf6c)

![ce4cab7ac4779ae679af09327fb7cccf](https://github.com/user-attachments/assets/4a389421-3ef2-471b-88fc-9942b7731b28)

和去年一样，还是拿小米 10 拍的，仍然没有换手机。




![a552b4b1fc39f851dfdb027be7a75d6b_720](https://github.com/user-attachments/assets/b261d236-fd1a-4a19-8764-e625c1dab92d)
最近看了《小市民系列 - 春季限定草莓塔事件》，挺有意思的。
原作主角两人很遗憾没有吃到草莓塔，所以我替他们吃了。

</details>

cfdxkk committed:
feat(privacy_visibility): user info visibility settings (store settings persistently).

可以持久化存储用户的信息可见性设置和关联平台可见性设置。

cfdxkk committed:
feat(feed): update video api for following function

更新视频 API 类型，支持获取用户是否已经被关注

cfdxkk committed:
fix(ssr): get correct following status in video page when SSR.

cfdxkk committed:
feat(following): follow in video page

1. 视频页的关注功能。
2. 关注功能的后端支持。

cfdxkk committed:
refactor(pinia): unify types for pinia `user-info` and API response.

1. 统一 pinia 的 `user-info` store 中的用户信息类型和 API 的用户自己的信息响应的类型。现在 API 响应什么 pinia 里就存储什么。
2. 其他 QoL 改进。

cfdxkk committed:
fix(login): wrong input autoComplete during 2FA login.

fix #292
修复了 2FA 时不正确的输入框自动填充

cfdxkk committed:
fix(user-page): make "follow" button in user-page work around.

fix #293

1. 修复了一个 SSR 时用户主页崩溃的问题（疑似请求结果导致的状态更新的副作用导致了无限循环）。
2. 为用户主页关注按钮添加实际功能，同时修复了在 SSR 时工作异常的问题。
3. 其它 QoL 更新。

cfdxkk committed:
fix(login): wrong input autoComplete during create 2FA.

fix #292
修复了 2FA 时不正确的输入框自动填充。
改为 `autocomplete="off"` 禁止自动填充了。

cfdxkk committed:
feat: unfollowing.

cfdxkk committed:
fix: exporting a type that was forgotten to be exported

cfdxkk committed:
Merge remote-tracking branch 'origin/develop' into feature-2025011201-Feed-cfdxkk

cfdxkk committed:
refactor(rbac): new way to check user role

hafterain committed:
refactor: font list adjustment

优化和精简字体设置。

- 注释文本：调整模糊不清（如拿操作系统指代字体）或错误的描述（如“Roboto Flex”和“可变Roboto”的混淆）。
- CJK大字符集：精简一般不会装的字体，优先使用操作系统字体。
    - SimSun-ExtB、SimSun-ExtG已覆盖B、C、D、E、F、G、H、I几个扩展区且随所有版本Windows分发，因此移除字形偏丑的MingLiU。
    - BabelStone项目将从BabelStone版本17开始分为三个独立文件BabelStone Han BMP、BabelStone Han SIP、BabelStone Han TIP，加入这些为未来做准备。BabelStone Han出于兼容原因保留。
    - 互联网上一部分文章仍推荐使用者安装花园明朝或字云，出于热门度和兼容原因保留。
- 回退：去除微软正黑体，质量低劣。
- 英文：去除过时、质量低下、无法直接调用的字体，微调字体顺序。
    - SF Pro无法直接调用；Oxygen老掉牙（KDE 4时代的东西）字重还不够；Cantarell即将被GNOME项目淘汰，由一Inter变体（Adwaita Sans，实际就Inter开了`ss05`）取代；Ubuntu字面太小，和中文混排效果太差。因此移除三者。
    - Fira Sans实际已停止更新，其LGC部分的bug（见其项目issues页面）再也不会得到修复，因此调整其回退顺序至Noto Sans之后。
- 简体中文：移除手机常用字体，调整字体回退顺序。
    - HarmonyOS Sans、MiSans、MILan Pro VF实际在大部分终端用户的计算机上上装机量较少，且国产Android设备的浏览器 / WebView一般直接无视CSS中的`font-family`（比较有自己的想法），因此也无需为它们准备回退。
    - 都macOS 15.3.1了，苹方还是不给非Safari浏览器开放其可变字重，因此将思源调整至最前——这实际是“双赢”选择，没装思源及iOS用户该用苹方还是用苹方。
- 台湾繁体中文：移除手机常用字体，调整字体回退顺序，移除日标冬青。
    - 日标字还是传统印刷风格，和台标冬青的“楷化印刷体”风格不搭，改为直接回退到字符集较大的国标。
- 香港繁体中文：移除手机常用字体，调整字体回退顺序，添加一个字重较多的日标冬青（Hiragino Sans，iOS / macOS设备可用，9字重）。
- 日语：添加一个字重较多的日标冬青（Hiragino Sans，iOS / macOS设备可用，9字重），移除所有中文冬青，移除MS Gothic。
    - 对于未安装日标冬青但安装了中文冬青的用户，原回退顺序将使它们显示中文风格的汉字，这对日语用户有些奇怪。
    - MS Gothic和SimHei / SimSun差不多都是老掉牙的字体，而且点阵字级范围比SimSun都大，实在没必要保留。
- 徽标：去除无法直接调用的SF Pro。
- 等宽：去除过时、质量低下、无法直接调用的字体。
    - Monaco十分老旧且且只有一个字重；Cousine实属丑陋；Segoe UI Mono、SF Mono无法直接调用。因此移除三者。

hafterain committed:
fix: update comments

hafterain committed:
fix: clean up unnecessary comments

Aira-Sakuranomiya committed:
fix: resolve autoprefixer not working (#300)

Aira-Sakuranomiya committed:
feat: new scrollbar (#302)

* feat(scroll container): new scrollbar component

* feat: add scroll test page

* refactor: use new scroll component

* feat: restore nuxt scroll behavior for custom element

Aira-Sakuranomiya committed:
feat(scroll container): better scrollbar logic

hafterain committed:
Update _variables.scss

hafterain committed:
remove segoe ui variable

otomad committed:
Update _variables.scss

hafterain committed:
refactor: add more undistinguish chatacters

hafterain committed:
Update username.pom refactor: add more undistinguish chatacters

otomad committed:
refactor: update username requirements

Accepts non leading, trailing, continuous spaces.

Close #286

otomad committed:
Merge pull request #305 from KIRAKIRA-DOUGA/refactoring-2025030701-UsernameRegex-hafterain

Refactoring 2025030701 username regex hafterain

otomad committed:
Merge pull request #304 from KIRAKIRA-DOUGA/refactoring-2025030701-UsernameRegex-hafterain

refactor: modify undistinguish characters

otomad committed:
Merge pull request #306 from KIRAKIRA-DOUGA/hotfix-2025031301-MailImage-otomad

feat: provide mail images with permanent link

hafterain committed:
Update _variables.scss

hafterain committed:
Update _variables.scss

otomad committed:
refactor: font list adjustment

Merge pull request #301 from KIRAKIRA-DOUGA/refactoring-20250211-FontListAdjustment-hafterain

otomad committed:
hotfix(crowdin): rosales locales are added to crowdin

otomad committed:
hotfix(crowdin): rosales locales are added to crowdin

otomad committed:
Update Crowdin configuration file

Aira-Sakuranomiya committed:
feat: global line-height

Aira-Sakuranomiya committed:
fix: resolve backdrop-filter not working during transition

父元素的opacity低于1，backdrop-filter会直接失效
Web，很神奇吧

cfdxkk committed:
fix: fix typo in CreationCommentsItem.vue

cfdxkk committed:
Merge pull request #299 from KIRAKIRA-DOUGA/feature-2025020901-RBAC-cfdxkk

✨ KIRAKIRA RBAC

cfdxkk committed:
Merge remote-tracking branch 'origin/develop' into feature-2025011201-Feed-cfdxkk

cfdxkk committed:
feat: update unfollowing success message display type

Aira-Sakuranomiya committed:
feat(button): add decoration for warning and danger

Aira-Sakuranomiya committed:
fix(capsule-slider): resolve backdrop-filter not working during transition

Aira-Sakuranomiya committed:
refactor: better looking chip-shadow

Aira-Sakuranomiya committed:
fix(scroll-container): fix visibility logic for scrollbars

otomad committed:
docs(readme): add architecture diagram

Aira-Sakuranomiya committed:
fix: resolve unexpected `/_icons` page

Aira-Sakuranomiya committed:
chore: remove useless app config

otomad committed:
fix(router): firefox cannot switch languages

otomad committed:
fix(router): type error

otomad committed:
fix(router): wrong firefox timeout

otomad committed:
try(router): fix language transition

Aira-Sakuranomiya committed:
fix: fix language transition

otomad committed:
refactor(router): remove duplicate code

写重复了。

otomad committed:
fix(no script): weird refresh icon positioning

Close #280 .

otomad committed:
feat: add akkarin avatar for guests

Closes #288 .

otomad committed:
refactor(about): simplify version tag

otomad committed:
feat: add new easings

Sync eslint and stylelint configs.

otomad committed:
feat(component): trans interpolation

Aira-Sakuranomiya committed:
build: try to fix build error

otomad committed:
feat(font): introduce yozora sans

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

cfdxkk committed:
feat: add daemon progress for home page video

Aira-Sakuranomiya committed:
feat: bigger akkarin avatar

Aira-Sakuranomiya committed:
feat(i18n): update english locale

Aira-Sakuranomiya committed:
feat(i18n): update locales

otomad committed:
feat(privacy): add social media platforms

Closes #311 .

otomad committed:
fix(git): stuck on commit

otomad committed:
fix(volar, eslint): fix eslint and vue official load failed

通过修改 tsconfig 修复了我这么多年一直没法用 vue 的 bug。

otomad committed:
refactor(index): remove comments

Aira-Sakuranomiya committed:
build: ensure package manager version is specified

otomad committed:
feat(i18n): introduce in context localization

Remove languages from i18n strings, and will use JS Intl API to get language names.

otomad committed:
refactor(icon): update regexp

上次艾拉直播时已经改了正则表达式的图标，但是就一直没放过来。

otomad committed:
fix(trans interpolation): fix bad regexp

Aira-Sakuranomiya committed:
refactor(i18n): replace hardcoded messages with localized strings (#312)

* refactor: replace some hardcoded messages with localized strings for i18n

* refactor: replace some hardcoded messages with localized strings for i18n

* refactor: replace some hardcoded messages with localized strings for i18n

* refactor: replace hardcoded messages with localized strings

* refactor: add to blocklist → block user

* refactor: update dev mode and replace hardcoded messages with localized strings

* refactor: better locale strings

* feat(i18n): update locales

* Update

* Update

---------

Co-authored-by: Rantetsu Inori <shuishenme@outlook.com>

otomad committed:
fix(crowdin): close icon cannot be seen clearly in dark theme

otomad committed:
fix(crowdin): crowdin phrase cannot be selected or clicked

otomad committed:
feat(i18n): update schinese and cantonese translations

otomad committed:
refactor(types): separate jipt type declarations

Aira-Sakuranomiya committed:
feat: better hover and focus

otomad committed:
refactor: optimize improve translation logo

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

otomad committed:
refactor: update eslint

otomad committed:
fix: typo

otomad committed:
refactor: rename the arg

Aira-Sakuranomiya committed:
feat(modal): enhance mobile usability

Aira-Sakuranomiya committed:
fix(modal): fix tiny titlebar when no close button

Aira-Sakuranomiya committed:
fix: discord server link

otomad committed:
docs(readme): update badges, add wiki link and contributors

Aira-Sakuranomiya committed:
refactor: remove css-doodle

Aira-Sakuranomiya committed:
feat(settings): update settings layout

Aira-Sakuranomiya committed:
refactor(api): remove show css doodle

cfdxkk committed:
feat: home page video blocklist (don't work in top-nav SSR)

otomad committed:
docs(readme): sublime text with webstorm link

cfdxkk committed:
feat(block): block page (in progress)

Aira-Sakuranomiya committed:
feat(settings): update language page

cfdxkk committed:
feat(block): block page (in progress)

Aira-Sakuranomiya committed:
refactor: favorites → collections

Aira-Sakuranomiya committed:
feat(i18n): update locales

Aira-Sakuranomiya committed:
fix(i18n): fix feed locale

Aira-Sakuranomiya committed:
refactor: change following feed url

Aira-Sakuranomiya committed:
feat: following feed page prototype

Aira-Sakuranomiya committed:
feat(modal): smoother mobile modal gesture

Aira-Sakuranomiya committed:
feat: update following feed

Aira-Sakuranomiya committed:
feat: add reached end to following feed

Aira-Sakuranomiya committed:
feat(offcanvas): make cursor pointer on offcanvas icons

for people who use a mouse on their phone haha

cfdxkk committed:
feat(block): block page (block-user) (in progress)

cfdxkk committed:
feat(block): block page (block-user) (in progress)

Aira-Sakuranomiya committed:
feat(alert): update alert layout

Aira-Sakuranomiya committed:
feat: update shadows

Aira-Sakuranomiya committed:
feat: pull to refresh

Aira-Sakuranomiya committed:
feat: hide banner on mobile devices

Aira-Sakuranomiya committed:
feat(user-content): update layout

Aira-Sakuranomiya committed:
feat: new upload page design

cfdxkk committed:
feat(block): block page (hide-user) (in progress)

Aira-Sakuranomiya committed:
fix: prevent unexpected autocomplete

otomad committed:
feat(nuxt): upgrade to 3.17.4

otomad committed:
fix(i18n): bad nuxt instance called in a composable

用了一些绕弯的方法又莫名其妙可以用了，不过我估计下次更新 Nuxt 又会出现新问题。

Aira-Sakuranomiya committed:
build: migrate to bun

otomad committed:
chore(deps): try bun install

otomad committed:
try(i18n): but not work

Aira-Sakuranomiya committed:
fix(i18n): resolve i18n not working

Aira-Sakuranomiya committed:
build: revert to node and pnpm

Aira-Sakuranomiya committed:
fix(i18n): resolve i18n not working

Aira-Sakuranomiya committed:
fix: shut up

Aira-Sakuranomiya committed:
fix(i18n): resolve i18n bugs

Aira-Sakuranomiya committed:
fix: fix `isInContextLocalization`

Aira-Sakuranomiya committed:
fix: shut build warning up

Aira-Sakuranomiya committed:
build: remove unnecessary packages and upgrade

Aira-Sakuranomiya committed:
perf(scroll-container): add passive to listener

Aira-Sakuranomiya committed:
fix: prevent unexpected autocomplete

auto-complete默认就是关的，之前密码乱补全是因为浏览器看到email和password类型就不管auto-complete了

Aira-Sakuranomiya committed:
fix: update local storage only on client

Aira-Sakuranomiya committed:
feat: add new icon

Aira-Sakuranomiya committed:
build: better developer experience

Aira-Sakuranomiya committed:
chore: remove disable format on paste

otomad committed:
chore(cover): add cover blank for certain purposes

otomad committed:
refactor(eases): change sassdoc easing formulas

Aira-Sakuranomiya committed:
build: fix packages not being built

Aira-Sakuranomiya committed:
feat: add `theme-color` based on whether colored sidebar is on

Aira-Sakuranomiya committed:
feat: better variable font transitions

Aira-Sakuranomiya committed:
build: better developer experience

Aira-Sakuranomiya committed:
feat(progress-ring): rounded indeterminate progress ring

otomad committed:
feat(player video slider): add more control buttons

1. 增加停止按钮、首帧按钮、末帧按钮。
2. 优化拖拽时间进度条逻辑，会自动暂停避免异常，如产生卡碟效果。
3. 新增允许拖拽时间进度条后自动恢复播放（即与B站行为一致，可选功能）。
4. 拖拽时间进度条时，播放/暂停按钮图标自动变为快进/快退图标。
5. 更改背景图片设置的着色和模糊强度图标。

Aira-Sakuranomiya committed:
feat: adjust player video panel settings layout

otomad committed:
fix(i18n): wrong type `t`

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

Aira-Sakuranomiya committed:
feat: new layouts

- 视频页布局大改。
- 在需要的时候隐藏AppBar和NavigationBar。
- 修正一些用语。

Aira-Sakuranomiya committed:
feat: display user roles

Aira-Sakuranomiya committed:
refactor: adjust details

Aira-Sakuranomiya committed:
refactor: adjust details

cfdxkk committed:
feat(block): block page (block-tag) (in progress)

otomad committed:
fix(components): useSlots() → $slots

明明有现成的 `$slots` 不用，非要额外写一个 `const slots = useSlots();` 再用新定义的 `slots`。咋了，冲代码行数业绩？

cfdxkk committed:
feat(block): block setting page

cfdxkk committed:
Merge remote-tracking branch 'origin/develop' into feature-2025011201-Feed-cfdxkk

cfdxkk committed:
fix(ssr): error in cookie-baker

修复了 SSR 时烘焙网络曲奇☆失败的问题。（Error: 烘焙网络曲奇☆时发生了意料外的错误）
cookieBaker 中 调用的 getSelfUserInfo 函数中使用的 🍍Pinia Store 会导致该错误，通过添加一个新的函数参数来显式指定是否使用 🍍Pinia Store 来解决

cfdxkk committed:
fix(lint): lint waring in UserController.ts

otomad committed:
refactor(i18n): blur intensity → blurriness

Aira-Sakuranomiya committed:
build: upgrade pnpm

Aira-Sakuranomiya committed:
fix(player-video): fix player bugs

Aira-Sakuranomiya committed:
fix(player-video): fix new options not saving

otomad committed:
feat(player): add replay button

otomad committed:
refactor(eslint): add max props length per line

otomad committed:
fix(eslint): forget first line prop

Aira-Sakuranomiya committed:
refactor: remove sharp and flat appearance mode

close #124

Aira-Sakuranomiya committed:
feat: update format icons

Aira-Sakuranomiya committed:
feat: better comments loading

Aira-Sakuranomiya committed:
fix: resolve hidden comments still clickable

cfdxkk committed:
fix: error in cookie-baker and block list related issue.

cfdxkk committed:
fix: update pnpm-lock.yaml

cfdxkk committed:
Merge remote-tracking branch 'origin/develop' into feature-2025011201-Feed-cfdxkk

cfdxkk committed:
fix(build): build error due to missing modules in pnpm-lock.yaml

Aira-Sakuranomiya committed:
style: resolve wrong page name format

Aira-Sakuranomiya committed:
fix(settings): fix transition not working on `block and hide` page

<template>必须只有一个根节点才能正常使用page transition

Aira-Sakuranomiya committed:
chore: better developer experience

Aira-Sakuranomiya committed:
feat: better following experience

otomad committed:
fix(button): correct way to get dom element ref

Aira-Sakuranomiya committed:
feat: follow button component

cfdxkk committed:
Merge pull request #294 from KIRAKIRA-DOUGA/feature-2025011201-Feed-cfdxkk

Privary(v1) & Feed & Following & block/hide

cfdxkk committed:
fix: remove useless output

otomad commented on README.md:
````diff
@@ -338,7 +421,7 @@ $t(variable)
 <td>
 
-```typescript
+```typescripthttps://img.shields.io
````
？？？全局替换错误了？

Aira-Sakuranomiya replied:
我猜是你手滑粘贴错位置了

otomad commented on .gitattributes:
```diff
@@ -26,7 +26,6 @@
 *.stylus text eol=lf
 *.pcss text eol=lf
 *.sss text eol=lf
-*.css-doodle text eol=lf linguist-language=CSS
```
此处不建议移除。
这个是 gitattributes，说实话不移除也不会影响。但万一以后打算重新加上，这个到时可能又会忘记加了。

Aira-Sakuranomiya replied:
已经移除了我觉得也没有必要加上，因为CSS Doodle我想不到任何可能再次引入的原因。

otomad commented on components/Checkbox.vue:
```diff
@@ -294,7 +305,9 @@
 	}
 
-	@each $key in "", "-back" { // 故意把动画写两遍，让 CSS 以为是两个动画。
+	@each $key in "", "-back" {
+
+		// 故意把动画写两遍，让 CSS 以为是两个动画。
```
奇怪究竟是哪条 stylelint 规则强制把注释分行写还前面加空行？

感觉像是某个自动格式化工具干的。

otomad commented on components/Checkbox.vue:
```diff
@@ -273,11 +280,15 @@
 		}
 
-		:comp.checked:focus & {
+		:comp:any-hover input:checked + &,
+		:comp:any-hover input:indeterminate + &,
+		:comp:focus-visible input:checked + &,
+		:comp:focus-visible input:indeterminate + & {
```
```suggestion
		:comp:is(:any-hover, :focus-visible) input:is(:checked, :indeterminate) + & {
```
注意，请务必测试，有小概率可能不能用，但绝对比现在的写法优雅许多。

otomad commented on components/Checkbox.vue:
```diff
@@ -273,11 +280,15 @@
 		animation: pressing-back $duration-half $ease-in alternate 2;
 
-		:comp:focus & {
+		:comp:any-hover &,
+		:comp:focus-visible & {
```
```suggestion
		:comp:is(:any-hover, :focus-visible) & {
```

otomad commented on components/Creation/CreationComments/CreationCommentsItem.vue:
```diff
@@ -1,4 +1,6 @@
 <script setup lang="ts">
+	import NumberFlow from "@number-flow/vue";
```
建议用自动全局导入的方式，不要每次都手动导入。

otomad commented on components/Creation/CreationDetail.vue:
```diff
@@ -1,4 +1,6 @@
 <script setup lang="ts">
+	import NumberFlow from "@number-flow/vue";
```
建议用自动全局导入的方式，不要每次都手动导入。

otomad commented on components/Logo/LogoSettings.vue:
```diff
@@ -11,7 +11,7 @@
 			<mask
 				id="mask0_3293_20384"
-				style="mask-type:alpha"
+				style="mask-type: alpha;"
```
```suggestion
				:style="{ maskType: 'alpha' }"
```
禁止使用字符串式行内样式。

Aira-Sakuranomiya replied:
原因是什么？这里没有任何必要这么做吧？

otomad replied:
> 原因是什么？这里没有任何必要这么做吧？

据我所知好像是字符串样式会被 Vue 二次解析，直接写成对象会省去解析这一步。

不过我是感觉全部写成对象形式更统一一致。

otomad commented on components/Logo/LogoSettings.vue:
```diff
@@ -23,23 +23,25 @@
+			<g class="circle">
+				<mask
+					id="mask1_3293_20384"
+					style="mask-type: alpha;"
```
```suggestion
					:style="{ maskType: 'alpha' }"
```
禁止使用字符串式行内样式。

Aira-Sakuranomiya replied:
原因是什么？这里没有任何必要这么做吧？

otomad replied:
> 原因是什么？这里没有任何必要这么做吧？

据我所知好像是字符串样式会被 Vue 二次解析，直接写成对象会省去解析这一步。

不过我是感觉全部写成对象形式更统一一致。

otomad commented on components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue:
```diff
@@ -1,9 +1,9 @@
 <script setup lang="ts">
+	import NumberFlow from "@number-flow/vue";
```
建议用自动全局导入的方式，不要每次都手动导入。

otomad commented on components/PullToRefresh.vue:
我很好奇为什么 props 要传入选择器而不是 HTML DOM ref？

另外，在 CSS 样式表中我完全没有看到 `overscroll-behavior` 这个属性的出现，不太清楚你是如何防止默认事件触发的。

Aira-Sakuranomiya replied:
因为没有便捷的方法拿到位于layout上的DOM ref。

默认事件触发这个我需要在ScrollContainer里解决，已解决。

otomad commented on components/PullToRefresh.vue:
```diff
@@ -0,0 +1,147 @@
+		/** 滚动容器 querySelector。 */
+		scrollContainerSelector?: string;
+		/** 滚动内容 querySelector。 */
+		scrollContentSelector?: string;
```
```suggestion
		/** 滚动容器选择器。 */
		scrollContainerSelector?: string;
		/** 滚动内容选择器。 */
		scrollContentSelector?: string;
```

otomad commented on components/PullToRefresh.vue:
```diff
@@ -0,0 +1,147 @@
+	});
+	const refreshing = defineModel<boolean>();
+
+	const scrollContainer = ref<HTMLElement | null>(null);
+	const scrollContent = ref<HTMLElement | null>(null);
```
```suggestion
	const scrollContainer = ref<HTMLElement>();
	const scrollContent = ref<HTMLElement>();
```
据我所知 null 是多余的，官网的教程多做的这一步是纯扯淡的。

Aira-Sakuranomiya replied:
不这么写类型会报错，你可以试试。

otomad replied:
> 不这么写类型会报错，你可以试试。

了解。

otomad commented on components/PullToRefresh.vue:
```diff
@@ -0,0 +1,147 @@
+		if (scrollContainer) {
+			scrollContainer.value?.addEventListener("touchstart", onTouchStart as EventListener);
+			scrollContainer.value?.addEventListener("touchend", onTouchEnd as EventListener);
+		}
```
你这段 `if (scrollContainer)` 判断了个寂寞啊，它恒为 true。

话说我封装了更强大的 useEventListener composable，为啥还要原始的 onMounted addEventListener + onBeforeUnmount removeEventListener？

```suggestion
		scrollContainer.value?.addEventListener("touchstart", onTouchStart as EventListener);
		scrollContainer.value?.addEventListener("touchend", onTouchEnd as EventListener);
```

Aira-Sakuranomiya replied:
> 话说我封装了更强大的 useEventListener composable，为啥还要原始的 onMounted addEventListener + onBeforeUnmount removeEventListener？

因为需要在选择器执行完毕后再执行

otomad commented on components/PullToRefresh.vue:
```diff
@@ -0,0 +1,147 @@
+		if (scrollContainer.value) {
+			scrollContainer.value.removeEventListener("touchstart", onTouchStart as EventListener);
+			scrollContainer.value.removeEventListener("touchend", onTouchEnd as EventListener);
+		}
```
话说我封装了更强大的 useEventListener composable，为啥还要原始的 onMounted addEventListener + onBeforeUnmount removeEventListener？

```suggestion
		scrollContainer.value?.removeEventListener("touchstart", onTouchStart as EventListener);
		scrollContainer.value?.removeEventListener("touchend", onTouchEnd as EventListener);
```

Aira-Sakuranomiya replied:
> 话说我封装了更强大的 useEventListener composable，为啥还要原始的 onMounted addEventListener + onBeforeUnmount removeEventListener？

因为需要在选择器执行完毕后再执行

otomad commented on components/PullToRefresh.vue:
```diff
@@ -0,0 +1,147 @@
+		<Transition>
+			<div v-if="pulling || refreshing" ref="indicatorWrapper" class="indicator-container" :style="{ transform: `translateY(${indicatorWrapperTranslateY}px)` }">
+				<div class="indicator">
+					<Icon v-if="!refreshing" name="refresh" :class="{ ready: (pullDistance >= REFRESH_THRESHOLD) }" :style="{ rotate: `${indicatorRotate}deg` }" />
```
```suggestion
					<Icon v-if="!refreshing" name="refresh" :class="{ ready: pullDistance >= REFRESH_THRESHOLD }" :style="{ rotate: `${indicatorRotate}deg` }" />
```

otomad commented on utils/regex.ts:
```diff
@@ -0,0 +1,33 @@
+ */
+export function isLooksLikeRegexString(str: string) {
+  // 1. 基本格式：以斜杠开头、斜杠结尾，可跟 0~6 位正则标志
+	const regexSyntax = /^\/(.+)\/([gimsuy]*)$/;
```
```suggestion
	const regexSyntax = /^\/(.+)\/([a-z]*)$/;
```

你永远不能预计未来 tc39 会增加哪些新的正则标志。
就比如你目前给的六位标志就不包含我最喜欢的 v 正则标志。

otomad commented on utils/imports.ts:
提示：将 NumberFlow 放在这里以自动导入。

```typescript
export { default as NumberFlow } from "@number-flow/vue";
```

otomad commented on types/modules.d.ts:
此处不建议移除。
理由同 .gitattributes。

Aira-Sakuranomiya replied:
已经移除了我觉得也没有必要加上，因为CSS Doodle我想不到任何可能再次引入的原因。

otomad commented on stores/app-settings.ts:
```diff
@@ -40,7 +37,9 @@ export const useAppSettingsStore = defineStore("app-settings", {
 				fontSizeScale: 1,
 			},
 			controller: {
+				showStop: false,
 				showFrameByFrame: false,
+				autoResumePlayAfterSeeking: false,
```
据我所知它缺少了 replay，看起来像是没同步。

otomad commented on plugins/scroll.client.ts:
你这算 nuxt 插件，应该放在 nuxt 子文件夹内。

plugins/nuxt/scroll.client.ts

otomad commented on pages/settings/privacy.vue:
不知道为什么感觉 visibilities 和 LINKED_ACCOUNT_VISIBILITIES_SETTING_ITEMS 是不是重复了？

otomad commented on pages/settings/dashboard.vue:
```diff
@@ -1,36 +1,36 @@
 	const selfUserInfoStore = useSelfUserInfoStore();
-	const userBirthdayDisplay = computed(() => selfUserInfoStore.birthday ? formatDateWithLocale(new Date(selfUserInfoStore.birthday)) : "Unknown");
-	const registerDateDisplay = computed(() => selfUserInfoStore.userCreateDateTime ? formatDateWithLocale(new Date(selfUserInfoStore.userCreateDateTime)) : "Unknown");
+	const userBirthdayDisplay = computed(() => selfUserInfoStore.userInfo.birthday ? formatDateWithLocale(new Date(selfUserInfoStore.userInfo.birthday)) : "Unknown"); // TODO: 生日功能
```
**讨论：目前的生日功能内部数据结构是什么？是否支持农历？**

YasaitoriOvO replied:
应该是一串时间戳 农历我认为可以用公历反推出来

otomad replied:
> 应该是一串时间戳 农历我认为可以用公历反推出来

反推倒也行，但是还是需要单独的一个数据来存储使用公历还是农历。

不过反推有一点需要注意，QQ 原本支持农历生日，新版本反而将其移除了，于是它近年总是在不是我生日的时候弹出生日提示，非常烦。

YasaitoriOvO replied:
> 但是还是需要单独的一个数据来存储使用公历还是农历。

这个我认为算是用户个性设置的一种，应该可以加个字段来判断这个

otomad commented on helpers/users.ts:
这是测试用户代码，现在已有真实后端的用户数据，感觉可以把这个文件删了。

otomad commented on components/Settings/SettingsChipItem.vue:
```diff
@@ -10,6 +10,8 @@
 		details?: Readable;
 		/** 尾随操作图标。 */
 		trailingIcon?: DeclaredIcons;
+		/** 是否禁止尾随操作图片点击 */
```
```suggestion
		/** 是否禁止尾随操作图片点击？ */
```

otomad commented on components/RadioButton.vue:
```diff
@@ -220,11 +227,13 @@
 		}
 
-		:comp:focus input:checked + & {
+		:comp:any-hover input:checked + &,
+		:comp:focus-visible input:checked + & {
```
```suggestion
		:comp:is(:any-hover, :focus-visible) input:checked + & {
```

otomad commented on components/RadioButton.vue:
```diff
@@ -220,11 +227,13 @@
 		animation: pressing-back $duration-half $ease-in alternate 2;
 
-		:comp:focus & {
+		:comp:any-hover &,
+		:comp:focus-visible & {
```
```suggestion
		:comp:is(:any-hover, :focus-visible) & {
```

otomad commented on components/Player/PlayerVideo/PlayerVideoPanel/PlayerVideoPanel.vue:
```diff
@@ -80,40 +36,32 @@
-							<TabItem id="danmaku-list">弹幕列表</TabItem>
+							<TabItem id="danmaku-list">{{ t(0).danmaku }}</TabItem>
 							<TabItem id="chapters">分段章节</TabItem>
 							<TabItem id="playlist">播放列表「」</TabItem>
```
意义不明的角单引号。

Aira-Sakuranomiya replied:
详见设计稿，这里计划中角单引号内是要显示播放列表的类型的，比如「分P」、「收藏夹」，不过其实可以直接显示在下方，可以去掉，已采纳。

otomad commented on .stylelintignore:
此处不建议移除。
此处以避免 stylelint 对 css-doodie 文件误解析，即便文件不存在。

Aira-Sakuranomiya replied:
已经移除了我觉得也没有必要加上，因为CSS Doodle我想不到任何可能再次引入的原因。

otomad reviewed (changes requested):
本次 pr 更改文件太多了，连 GitHub 都开始卡顿了。

YasaitoriOvO commented:
在视频页刷新时还是会报500，

YasaitoriOvO commented:
# 🧪测试报告
尽量按照 **测试焦点** 进行的测试 这里只提到出现的问题

### 一、📺视频页
1. 在视频评论区会同时出现两个翻页器，不论长短
![image](https://github.com/user-attachments/assets/e246a082-6cf7-479f-a822-1bc0b09c0a39)

---

2. ~被其他用户屏蔽后打开其他用户(执行屏蔽操作的用户)的视频提示异常~
> 我觉得这里打开后应跳转到[视频消失页](https://kirakira.moe/error/301)

![image](https://github.com/user-attachments/assets/ab03987e-17f8-4318-a625-106f591d218d)

### 二、🚫屏蔽页
1. 屏蔽用户的模态框内用户信息不显示
> 这个问题发生在其他用户将用户屏蔽后 执行屏蔽操作的用户再去对已经将他屏蔽的用户屏蔽的情况
> 我觉得这里可以根据向后端/info?uid=xxx ```isBlocked``` ```isBlockedByOther``` ```IsHidden``` 等三个字段来判断不弹出模态框

![image](https://github.com/user-attachments/assets/a2de3c45-d40d-4a9a-b3ff-9830a4cec330)

---

2. ~输入框输入很卡~
> 不知道GIF能不能看出来很卡

**Update: 这里是所有输入框的通病**

![ezgif-69ca26f601f418](https://github.com/user-attachments/assets/bacc7c3e-21a6-4976-96eb-a1b0447e609c)

3. 没有对已屏蔽的用户做判断
> 前端这里没有对已屏蔽的用户做判断，还是会向后端发送请求

![image](https://github.com/user-attachments/assets/b1e6bdc2-f89e-4765-a1b3-9e3e812a1db0)

### 三、🤵用户页
1. ~被其他用户屏蔽后打开其他用户的主页为空~
> 我的建议是 一(2) 同理

2. #317 还没修复

## ❓疑问和一些话

1. 对于 **一(1)** 这个Bug(?)是刻意留下的吗，我有些不确定
3. ~02应该会对 **二(2)** 有点印象 但 _已知问题_ 没提到 那我这里再提下~
3. 粉丝数显示是打算包含在下一个更新里吗？

---
还会更新 最近更新：2025/7/6

<details> <summary>闲话</summary>
这几天去南京玩，明明没去几个地方人快累死了天呐

欢乐谷还给我送了张重玩票 但我那天要走了:(
^^^^ (这里一个下午比我前两天加起来还累）

乳糖不耐受还去喝奶茶 真是佩服我自己

PS. 附一张音乐台的照片，来晚了 鸽子跑走了
![IMG_20250625_105919](https://github.com/user-attachments/assets/d0601982-e734-4283-aee3-106e5b975f8e)
</details>

otomad commented:
回复 @YasaitoriOvO 的测试报告

---

1. 两个翻页器
这个我推测是用户有可能位于评论区的顶端或底端，如果位于评论区顶端则可以使用顶端的翻页器，反之亦然，有可能是方便用户操作翻页更方便。
2. 被屏蔽后提示视频异常
“屏蔽”我没理解错的话应该是“黑名单”功能，不是“隐藏”功能。我认为用户被屏蔽后，绝大多数功能应该不受任何影响（包括访问用户页、观看视频等、甚至是关注），唯一影响的是互动功能（如评论视频、私信等）。因为用户可以在游客（即未登录）情况下访问被屏蔽人的内容，因此返回任意错误（我觉得非要找个错误那么最贴切的错误是403，即无权限）没有任何意义。我觉得最好的方式是用户完全无感，就好像和没有屏蔽是一样的，只是所有评论、私信等互动会被系统默默吞掉，没有任何提示。

YasaitoriOvO commented:
> 1. 两个翻页器
>这个我推测是用户有可能位于评论区的顶端或底端，如果位于评论区顶端则可以使用顶端的翻页器，反之亦然，有可能是方便用户操作翻页更方便。

这可以通过计算评论内容的长度（或数量）或用户的可视区域范围，来实现仅在可视区域内显示一个翻页器吗？


> 2. 被屏蔽后提示视频异常
>    “屏蔽”我没理解错的话应该是“黑名单”功能，不是“隐藏”功能。我认为用户被屏蔽后，绝大多数功能应该不受任何影响（包括访问用户页、观看视频等、甚至是关注），唯一影响的是互动功能（如评论视频、私信等）。因为用户可以在游客（即未登录）情况下访问被屏蔽人的内容，因此返回任意错误（我觉得非要找个错误那么最贴切的错误是403，即无权限）没有任何意义。我觉得最好的方式是用户完全无感，就好像和没有屏蔽是一样的，只是所有评论、私信等互动会被系统默默吞掉，没有任何提示。

我的理解是视频依旧可以观看，但是所有互动执行但不向后端发送请求，发送请求也是返回错误这么个意思是吧

otomad commented:
> 1. 这可以通过计算评论内容的长度（或数量）或用户的可视区域范围，来实现仅在可视区域内显示一个翻页器吗？

确实，用 sticky 把翻页器固定在顶部或底部更合适（类似于用户页的选项卡）。

> 2. 我的理解是视频依旧可以观看，但是所有互动执行但不向后端发送请求，发送请求也是返回错误这么个意思是吧

我感觉应该是前端默认发送成功，比如发送一条评论或弹幕，此时提示发送成功并且立即显示在评论/弹幕列表中，但是一刷新网页就又消失了。至于后端，我感觉也可以做成全部返回成功，但是实际上后端不做任何处理。

nuxthub-admin[bot] committed:
ci: add nuxthub workflow

Aira-Sakuranomiya committed:
try: nuxt hub

Aira-Sakuranomiya committed:
Revert "ci: add nuxthub workflow"

This reverts commit 0c442aa12bef40d32bbe046a2d82d639656444dd.

Aira-Sakuranomiya committed:
Revert "try: nuxt hub"

This reverts commit 0c12ac6b158c0fdbd06423ece966820f5fa44460.

Aira-Sakuranomiya committed:
docs: fix a bug

Aira-Sakuranomiya committed:
build: auto import number flow

Aira-Sakuranomiya commented:
还有个问题，自从加了拖动时按钮图标改变之后，移动端拖动会出现一直疯狂seeking的问题就又出现了

Aira-Sakuranomiya committed:
fix(scroll-container): optimize and fix bugs

Aira-Sakuranomiya committed:
style: move scroll plugin

Aira-Sakuranomiya committed:
delete: remove temp user info helper

cfdxkk committed:
fix: hotfix danmaku module not work in SSR

hotfix danmaku module not work in SSR, but idk why...
`environment.client` is not working but `import.meta.client` work very well...

cfdxkk commented:
>“屏蔽”我没理解错的话应该是“黑名单”功能，不是“隐藏”功能。我认为用户被屏蔽后，绝大多数功能应该不受任何影响（包括访问用户页、观看视频等、甚至是关注），唯一影响的是互动功能（如评论视频、私信等）。因为用户可以在游客（即未登录）情况下访问被屏蔽人的内容，因此返回任意错误（我觉得非要找个错误那么最贴切的错误是403，即无权限）没有任何意义。我觉得最好的方式是用户完全无感，就好像和没有屏蔽是一样的，只是所有评论、私信等互动会被系统默默吞掉，没有任何提示。

@otomad 会改成被屏蔽后可以正常访问屏蔽者的主页或视频。但无感发送评论这个我觉得不好，被屏蔽了就应该明确通知，我会增加字段让访问一个人的主页/视频时直接能够得知是否被屏蔽。以免出现打了一大段话之后发现发送失败。

cfdxkk committed:
feat: update block and hide function.

Update block and hide function.
1. User can visit the video and user info who block him.
2. User can't edit the comment and danmaku under the video if user is blocked by uploader.
3. Fix you are unable to block a user if this user is already blocked you.

cfdxkk commented:
”您已被该用户屏蔽，无法发送评论/弹幕“
![image](https://github.com/user-attachments/assets/caa3855c-b505-4c8f-8b80-153e2939eb65)

cfdxkk commented:
> 2. 输入框输入很卡

@YasaitoriOvO 在本地 dev 时比较卡，build 之后就好很多。而且这个地方我已经做了防抖，正则性能差也没办法了，只能这样了。

YasaitoriOvO committed:
fix: add debounce to input event handler

YasaitoriOvO commented:
> ”您已被该用户屏蔽，无法发送评论/弹幕“

@cfdxkk 

使用颜文字按钮依旧可以修改内容
![image](https://github.com/user-attachments/assets/00530ad5-98dd-4f9f-abf5-71fd9ee303da)

同时评论的颜文字按钮会使发送按钮可用
![image](https://github.com/user-attachments/assets/55434c91-b2b6-44a7-85bb-31d2bef50703)

YasaitoriOvO committed:
fix: correct dashboard display

Aira-Sakuranomiya committed:
feat(toggle switch): add shadow to toggle switch base

cfdxkk committed:
fix: kaomoji still available after blocked by others.

cfdxkk commented:
> 使用颜文字按钮依旧可以修改内容
> 同时评论的颜文字按钮会使发送按钮可用

@YasaitoriOvO Fixed

otomad committed:
refactor(error 301): disable auto play music

Close #321 .

cfdxkk commented:
> 被其他用户屏蔽后打开其他用户(执行屏蔽操作的用户)的视频提示异常~
> 我觉得这里打开后应跳转到[视频消失页](https://kirakira.moe/error/301)

@YasaitoriOvO Fixed (temp)
点击一个已屏蔽的视频或用户后，会跳转到 404 错误页面。（可能有潜在的更符合的错误页面，但暂时统一使用 404 页面以实现基本功能。）

cfdxkk committed:
feat: jump to 404 page if click a blocked video/user

1. 点击一个已屏蔽的视频或用户后，会跳转到 404 错误页面。（可能有潜在的更符合的错误页面，但暂时统一使用 404 页面以实现基础功能。）
2. 修复了无法查看屏蔽你的用户主页中的投稿的问题。

otomad committed:
style: install eslint-plugin-jsdoc

原被 eslint 弃用的 jsdoc 规则现已通过插件形式回归。目前的规定是，虽不要求必须写 jsdoc，但是如果写了 jsdoc，就必须要写规范。

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

otomad committed:
feat: use nuxt picture to display avif images

实验性功能：目前将所有的 NuxtImg 元素全部替换为了 NuxtPicture 元素。这样就可以自动显示文件大小更小的 avif 或 webp 格式图片。但可能会有部分布局异常，需手工调整。

otomad committed:
refactor: dropped a declaration from jsdoc

Aira-Sakuranomiya committed:
refactor(player video): remove player beeps

Aira-Sakuranomiya committed:
fix(player video): save settings correctly

YasaitoriOvO committed:
docs: fix some bugs

otomad committed:
refactor(animation): new language transition animation

Aira-Sakuranomiya commented:
@cfdxkk 我认为可以准备合并了，剩下的都是不大或者一直以来都有的问题了吧？

otomad committed:
fix: this.i18nKey is not a function

Not sure if it can work.

otomad committed:
feat(offcanvas): new offcanvas by design

使用自定义背景图片时，侧滑导航会有特殊效果。

otomad committed:
feat(base date time picker): start to add base date time picker

cfdxkk committed:
docs: remove useless console output

otomad committed:
Revert "feat(offcanvas): new offcanvas by design"

This reverts commit 15dc014e44efd092c8cfde630bc934704e5b8545.

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

otomad committed:
feat(timecode picker): finish timecode picker

Aira-Sakuranomiya committed:
feat(i18n): more i18n

Aira-Sakuranomiya committed:
build: upgrade pnpm

Aira-Sakuranomiya committed:
feat(i18n): update locales

Aira-Sakuranomiya committed:
feat(i18n): update locales

otomad committed:
feat: finish timecode picker and time picker

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

Aira-Sakuranomiya committed:
feat(i18n): update locales

Aira-Sakuranomiya committed:
fix: resolve #327

close #327

Aira-Sakuranomiya committed:
feat: better mobile experience

Aira-Sakuranomiya committed:
feat: optimize time display

- replace ratio signs with colon signs
- turn on font "case" feature for times

Aira-Sakuranomiya committed:
revert: revert auto import number flow

Aira-Sakuranomiya committed:
fix(slider): resolve slider bugs

fix(slider): resolve tooltip will not be displayed at correct position if number was NaN
fix(slider): resolve ssr error

YasaitoriOvO committed:
fix: sync dto

Aira-Sakuranomiya committed:
feat: use official transition interpolation component

Aira-Sakuranomiya committed:
test(i18n): disable i18n always redirect

Aira-Sakuranomiya committed:
revert(i18n): revert disable i18n always redirect

Aira-Sakuranomiya committed:
feat: elegant ripple

别突然加速，在手机上因为性能没那么好看起来会贼卡

Aira-Sakuranomiya committed:
feat(i18n): enable `typedOptionsAndMessages` for i18n

otomad committed:
fix: normalize user info

Resolve https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Rosales/issues/52 in frontend part.

otomad committed:
feat(date time picker): spining animation

cfdxkk committed:
feat: restore some functions in admin setting pages.

otomad committed:
feat: finish date picker

otomad committed:
feat: support relative date time

otomad committed:
Merge branch 'develop' of https://github.com/KIRAKIRA-DOUGA/KIRAKIRA-Cerasus into develop

otomad committed:
refactor(duration): use native intl duration format

虽然兼容性不佳，但是也使用旧的私有实现作为回退处理。

otomad committed:
fix(relative time): unreasonable behaviour

支持显示“昨天”“现在”而不是“1天前”“0秒前”。

otomad committed:
feat: add date picker to user info birthday settings

Optimize menu width changing animation.

cfdxkk committed:
feat: sync latest API data type.

Aira-Sakuranomiya committed:
chore(i18n): rename legacy component

Aira-Sakuranomiya committed:
feat(i18n): better translation interpolation component

Aira-Sakuranomiya committed:
feat: make dev tools happy
